addons/mail/static/src/core/web/discuss_client_action.js

```diff
@@ -30,6 +30,19 @@ export class DiscussClientAction extends Component {
             // bracket to avoid blocking rendering with restore promise
             this.restoreDiscussThread(nextProps);
         });
+        useEffect(
+            (thread) => {
+                if (thread) {
+                    const activeId =
+                        typeof thread.id === "string"
+                            ? `mail.box_${thread.id}`
+                            : `discuss.channel_${thread.id}`;
+                    this.props.updateActionState({ active_id: activeId });
+                    this.props.action.context.active_id = activeId;
+                }
+            },
+            () => [this.store.discuss.thread]
+        );
```

Comment:
If possible I would put logic in `restoreDiscussThread`, to avoid introducing a `useEffect` and have syncing issues with the other LOCs also related to change of discuss thread

Reply:
`updateActionState` is a prop that is only given to the action component. :grimacing: Also, we have the guarantee that discuss is the current action here while it's not the case in the model.

Reply:
Following the talk we had yesterday, `restoreDiscussThread` is only called when the action changes while we want to update the action anytime the active thread changes. So that's not the correct place unfortunately.